drivers/kscan/Kconfig.matrix

```diff
@@ -0,0 +1,84 @@
+# Keyboard scan matrix configuration options
+
+# Copyright (c) 2022 Nuvoton Technology Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+config KSCAN_MATRIX
+	bool "keyboard scan matrix"
+	help
+	  A matrix circuit is used in most computer keyboards where the key
+	  switches are connected by a grid of wires, similar to a diode matrix.
+
+if KSCAN_MATRIX
+
+config KSCAN_MATRIX_MAX_COLUMNS
+	int "keyboard matrix max column"
+	default 18
+	help
+	  Maximum columns of the matrix circuit.
+
+config KSCAN_MATRIX_MAX_ROWS
+	int "keyboard matrix max row"
+	default 8
+	help
+	  Maximum rows of the matrix circuit.
```

Comment:
why isn't this described in Devicetree? What if there are 2 devices with different configs?

Reply:
This follows the existing drivers which support the keyboard matrix hardware in the SoC.
As the keyboard matrix requires a lot of pads for column/row, most EC chips support only one hardware module from a cost perspective.